src/gui/tray/Window.qml

```diff
@@ -720,43 +720,45 @@ Window {
-                                if (actionList.length <= maxButtons) {
-                                    return actionList;
-                                }
-
-                                // add index to every action
-                                var actionsWithIndices = actionList.map(function(action, index) {
+                            Menu {
+                                id: moreActionsButtonContextMenu
+                                anchors.centerIn: parent
 
-                                    var actionWithIndex = { actionIndex: index };
+                                // transform model to contain indexed actions with primary action filtered out
+                                function actionListToContextMenuList(actionList, skipTransform) {
+                                    // early out with non-altered data
+                                    if (skipTransform === true) {
+                                        return actionList;
+                                    }
 
-                                    Object.assign(actionWithIndex, action)
+                                    // add index to every action
+                                    var actionsWithIndices = actionList.map(function(action, index) {
 
-                                    return actionWithIndex;
-                                });
+                                        var actionWithIndex = { actionIndex: index, label: action.label, primary: action.primary };
 
-                                // filter 'primary' action out
-                                var filteredActions = actionsWithIndices.filter(function(action) {
-                                    return action.primary !== true;
-                                });
+                                        return actionWithIndex;
+                                    });
 
-                                return filteredActions;
-                            }
+                                    // filter 'primary' action out
+                                    var filteredActions = actionsWithIndices.filter(function(action) {
+                                        return action.primary !== true;
+                                    });
 
-                            Menu {
-                                id: moreActionsButtonContextMenu
-                                anchors.centerIn: parent
+                                    return filteredActions;
+                                }
 
                                 Repeater {
                                     id: moreActionsButtonContextMenuRepeater
-                                    model: moreActionsButtonContextMenuContainer.actionListToContextMenuList(activityItem.links, activityListView.maxActionButtons)
+
+                                    readonly property bool skipTransformModel: activityItem.links.length <= activityListView.maxActionButtons

```

Comment:
Hm, why not have this check done in the function? This way the second parameter could be removed.